vertical-pod-autoscaler/enhancements/8818-in-place-only/README.md

```diff
@@ -0,0 +1,327 @@
+- Long-running computations: Jobs or services performing computations that cannot be checkpointed and would need to restart from the beginning.
+- Strict SLO requirements: Services with stringent availability requirements where even brief disruptions are unacceptable.
+In these scenarios, users would prefer:
+
+- To operate with current (potentially suboptimal) resource allocations until an in-place update becomes feasible
+- To receive clear signals when updates cannot be applied
+- To have VPA continuously retry updates as cluster conditions change
+
+## Goals
+
+- Provide a truly non-disruptive VPA update mode that never evicts pods
+- Allow VPA to eventually apply updates when cluster conditions improve
+- Respect the existing in-place update infrastructure from AEP-4016
+
+## Non-Goals
+
+- Guarantee that all updates will eventually succeed (node capacity constraints may prevent this)
+- Provide mechanisms to automatically increase node capacity to accommodate updates
+- Change the behavior of existing update modes (Off, Recreate, InPlaceOrRecreate)
+- Eliminate all possible disruption scenarios (see [Risk Mitigation](#risk-mitigation) for details on memory limit downsizing risks)
+
+
+## Proposal
+
+Add a new supported value of UpdateMode: `InPlace`
+This mode will:
+- Apply recommendations during pod admission (like all other modes)
+- Attempt in-place updates for running pods under the same conditions as `InPlaceOrRecreate`
+- Never add pods to `podsForEviction` if in-place updates fail
+- Continue monitoring for recommendation changes and re-attempt updates when new recommendations differ from the current `spec.resources` (see [Resize Status Handling](#resize-status-handling) for details)
+
+## Design Details
+
+Add `UpdateModeInPlace` to the VPA types:
+
+```golang
+// In pkg/apis/autoscaling.k8s.io/v1/types.go
+const (
+    // ... existing modes ...
+    // UpdateModeInPlace means that VPA will only attempt to update pods in-place
+    // and will never evict them. If in-place update fails, VPA will rely on
+    // Kubelet's automatic retry mechanism.
+    UpdateModeInPlace UpdateMode = "InPlace"
+)
+```
+
+### Infeasible Attempt Tracking
+
+VPA must track `infeasible` resize attempts to prevent infinite retry loops. This is necessary because infeasibility can be detected at different points depending on the Kubernetes version:
+| Kubernetes Version | When Infeasibility Is Detected                     | `spec.resources` After Attempt | How VPA Learns            |
+|--------------------|----------------------------------------------------|--------------------------------|---------------------------|
+| < 1.36             | After patch succeeds, kubelet reports status       | Updated to attempted value     | Resize status = Infeasible|
+| >= 1.36            | At patch time, API server rejects                  | Unchanged (old value)          | Patch error response      |
```

Comment:
Is this change guaranteed to land in 1.36? Adding this to the AEP right now may be a little early?